### PR TITLE
Copter: Add FRAME_TYPE to distinguish multicopters and helicopters

### PR DESCRIPTION
**Pull request by fhedberg**
Add FRAME_TYPE to distinguish multicopters and helicopters. This makes it easier to maintain other helicopter frames out of master (see #1860, #3245, etc.).

### CODE REVIEW

**Comment by R-Lefebvre** [comment]
Simple semantic change.  I like it.  "NotHelicopter" is just awful to read.

Does this work with the build system?

**Comment by fhedberg** [comment]
@r-lefebvre Yes

**Comment by fhedberg** [comment]
@rmackay9 @tridge Ping

**Comment by tridge** [comment]
I'm fine with this change. I don't find the different in #if all that much clearer, but I don't think its bad, and if it makes maintainence of patches by @fhedberg easier then it would be a good thing
Randy?

File: ArduCopter/defines.h [hunk 1]
```diff
@@ -83,6 +83,11 @@ enum aux_sw_func {
 #define SINGLE_FRAME 8
 #define COAX_FRAME 9
 
+// Frame types
+
+#define MULTICOPTER 0
```
Comment (magicrub) [inline ArduCopter/defines.h]: Wouldn't this be better named frame_multicopter?

**Comment by fhedberg** [comment]
Just to be overly clear; why this is nice is that adding a new helicopter frame (which I have a couple) can be done in one place and not in every single #if HELI... statement.

**Comment by fhedberg** [comment]
@rmackay9 Ping

**Comment by rmackay9** [comment]
Ok, i'll have a look, hopefully tomorrow..

**Comment by amilcarlucas** [comment]
Doesn't FRAME_CLASS do the same ??

**Comment by rmackay9** [comment]
@amilcarlucas the FRAME_CLASS parameter is only available after compilation though.  It might be possible to have a single firmware that handles helicopters and multicopters and then replace all the #if with checks of the parameter value but we're not quite there yet.

**Comment by khancyr** [comment]
@OXINARF @rmackay9 this can be closed no ?

**Comment by OXINARF** [comment]
@fhedberg Do you want to rebase this or is it surpassed now that multicopter is a single binary? You may still want this to have different heli frame macros, I'm not sure.

**Comment by davidbuzz** [comment]
no response back from @fhedberg in 5 months, PR is over a year old.

**Pull request closed without merging**